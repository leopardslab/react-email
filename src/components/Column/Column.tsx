import { CSSProperties, ReactNode } from 'react';
import { makeStyles } from '../../utils/makeStyles';
import { sx } from '../../utils/sx';
import type { BaseStyleProp } from '../types';
import { useTheme } from '../ThemeProvider';

export type ColumnStyles = 'root';

export interface ColumnProps extends BaseStyleProp<ColumnStyles> {
  children?: ReactNode;
  align?: 'left' | 'center' | 'right';
  verticalAlign?: CSSProperties['verticalAlign'];
}

const useStyles = makeStyles({
  root: {},
});

export const Column = ({
  children,
  className,
  classes,
  align = 'center',
  verticalAlign = 'top',
}: ColumnProps): JSX.Element => {
  const themeClasses = useTheme('column');
  const styles = useStyles({ classes }, themeClasses);

  return (
    <td className={className} style={sx(styles.root, { verticalAlign })} align={align}>
      {children}
    </td>
  );
};
