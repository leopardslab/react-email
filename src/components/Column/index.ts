export { Column, ColumnProps } from './Column';
