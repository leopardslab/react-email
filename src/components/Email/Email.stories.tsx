import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from './Email';

export default {
  component: Email,
} as ComponentMeta<typeof Email>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Email> = (args) => <Email {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: (
    <table style={{ width: '100%' }}>
      <tbody>
        <tr>
          <td>
            <p style={{ margin: '0', fontSize: '30px' }}>Hello World</p>
          </td>
        </tr>
      </tbody>
    </table>
  ),
  classes: {
    root: {
      backgroundColor: 'gray',
      color: 'white',
    },
  },
};
