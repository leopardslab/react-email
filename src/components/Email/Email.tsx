import { ReactNode } from 'react';
import { BaseStyleProp } from '../types';
import { makeStyles } from '../../utils/makeStyles';

type EmailStyles = 'root';

export interface EmailProps extends BaseStyleProp<EmailStyles> {
  children?: ReactNode;
}

const useStyles = makeStyles({
  root: {
    margin: '0px auto',
    maxWidth: '600px',
  },
});

export const Email = ({ children, className, classes }: EmailProps): JSX.Element => {
  const styles = useStyles({ classes });

  return (
    <div style={styles.root} className={className}>
      {children}
    </div>
  );
};
