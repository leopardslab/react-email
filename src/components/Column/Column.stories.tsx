import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from '../Email/Email';
import { Column } from './Column';

export default {
  component: Column,
} as ComponentMeta<typeof Column>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Column> = (args) => (
  <Email>
    <Column {...args} />
  </Email>
);

export const Default = Template.bind({});

Default.args = {
  children: (
    <tr>
      <td style={{ fontSize: '30px', color: 'red' }}>Hello World</td>
    </tr>
  ),
};
