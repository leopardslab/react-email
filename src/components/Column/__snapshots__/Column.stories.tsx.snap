// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Column Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="background-color: red; color: white; vertical-align: top;"
        >
          <p style="margin: 0px; font-size: 30px;">
            Hello World
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
